src/App.tsx

```diff
@@ -15,36 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  None = 'none',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+  Reverse = 'reverse',
+}
+
+type GoodsSortParams = {
+  sortField: SortType;
+  reversed: boolean;
+};
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, reversed }: GoodsSortParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabetical:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.None);
+  const [reversed, setReversed] = useState<boolean>(false);
+  const [initialGoods] = useState(goodsFromServer);
+
+  const visibleGoods = getPreparedGoods(initialGoods, { sortField, reversed });
```

Comment:
```suggestion
  const visibleGoods = getPreparedGoods(goodsFromServer, { sortField, reversed });
```

Why do you need this?